refactor/test: update test-suite

safesurfer opened the pull request:
update test-suite

safesurfer committed:
refactor/test: update test-suit

update test-suit

maidsafe-highfive commented:
r? @krishnaIndia

(maidsafe_highfive has picked a reviewer for you, use r? to override)

joshuef commented on web_hosting_manager/app/components/app_logs.js:
```diff
@@ -14,27 +14,27 @@ export default class AppLogs extends Component {
-    this.setState({loading: true});
-    this.props.getLogFilePath()
-      .then(path => {
+    this.setState({ loading: true }); // eslint-disable-line react/no-did-mount-set-state
```
instead of disabling linting here, can we rework this so that it's meeting the linter's requirements.

Do we need to be setting state in mount? Could it happen in the constructor?

Should we maybe be triggering global `redux` state changes instead of local to the component? This is afterall some part of the whole app state, right?

safesurfer replied:
Thank you josh. I'm updating the correction and will push it.

joshuef commented on web_hosting_manager/app/components/app_logs.js:
```diff
@@ -14,27 +14,27 @@ export default class AppLogs extends Component {
-    this.props.getLogFilePath()
-      .then(path => {
+    this.setState({ loading: true }); // eslint-disable-line react/no-did-mount-set-state
+    this.props.getLogFilePath() // eslint-disable-line react/prop-types
```
let's add this to a `propTypes` declaration for clearer errors etc, and then the linter will be happy too.

joshuef commented on web_hosting_manager/app/components/app_logs.js:
```diff
@@ -14,27 +14,27 @@ export default class AppLogs extends Component {
-      .then(path => {
+    this.setState({ loading: true }); // eslint-disable-line react/no-did-mount-set-state
+    this.props.getLogFilePath() // eslint-disable-line react/prop-types
+      .then(path => { // eslint-disable-line promise/always-return
```
here we could simply `return` inside the promise to keep to linter guides.

safesurfer replied:
updated it josh..

joshuef reviewed (changes requested):
Few notes on linter evasion here, @safesurfer

safesurfer committed:
minor refactor and update version of safe-node-app to latest version 0.5.3

safesurfer committed:
revert back the version change

safesurfer committed:
Merge branch 'test-suit' of https://github.com/safesurfer/safe_examples into safesurfer-test-suit

bochaco commented on web_hosting_manager/app/safenet_comm/temp.js:
```diff
@@ -1,11 +1,10 @@
 import temp from 'temp';
 
-const tempTrack = temp.track();
```
Are we sure we don't want to invoke `temp.track()` and explcitly removing it? 
From looking at the doc it seems that's to automatically track and cleanup files at exit: https://www.npmjs.com/package/temp

bochaco commented on web_hosting_manager/app/yarn.lock:
```diff
@@ -2,13 +2,14 @@
-"@maidsafe/safe-node-app@0.5.1":
-  version "0.5.1"
-  resolved "https://registry.yarnpkg.com/@maidsafe/safe-node-app/-/safe-node-app-0.5.1.tgz#79438d74a353408e3bf5e89ced5a462f0f384aa2"
+"@maidsafe/safe-node-app@0.5.3":
```
We should be using v0.6.0 already

bochaco commented on web_hosting_manager/app/components/app_logs.js:
```diff
@@ -13,28 +12,30 @@ export default class AppLogs extends Component {
         this.readLogFile();
-        this.setState({loading: false});
+        this.setState({ loading: false });
+        return Promise.resolve(true);
```
I think we've moved to use async/await so we should keep consistency here as well.

safesurfer replied:
I've updated the code gabriel..

safesurfer committed:
Merge branch 'master' of https://github.com/maidsafe/safe_examples into safesurfer-test-suit

safesurfer committed:
Merge branch 'master' of https://github.com/maidsafe/safe_examples into safesurfer-test-suit

hunterlester committed:
chore/config: update Node JS version requirement to ^8.0.0

safesurfer committed:
refactor/test: migrate from jest to mocha

migrate from jest to mocha

safesurfer committed:
migrate from jest to mocha and correct lints

safesurfer committed:
review corrections

hunterlester committed:
chore/update: update @maidsafe/safe-node-app to v0.6.1

hunterlester committed:
refactor/snippets: removes need to use new Uint8Array

bochaco committed:
Merge pull request #346 from hunterlester/update

chore/config: update Node JS version requirement to ^8.0.0

hunterlester committed:
refactor/snippets: removes unnecessary use of fromCharCode

bochaco committed:
Merge pull request #350 from hunterlester/playground

refactor/snippets: removes need to use `new Uint8Array`

safesurfer commented:
@joshuef @bochaco , I've updated the PR.

hunterlester committed:
refactor/styles: implement Shona and Shankar's CSS framework

bochaco committed:
Merge pull request #353 from hunterlester/design

refactor/styles: implement Shona and Shankar's CSS framework

hunterlester committed:
chore/dependency: updates electron to 1.7.11

hunterlester committed:
chore/dependency: updates @maidsafe/safe-node-app to v0.7.0

hunterlester committed:
chore/deps: updates CHANGELOG and includes updated lock files

bochaco committed:
Merge pull request #355 from hunterlester/electron

chore/dependency: updates electron to 1.7.11

safesurfer committed:
style/correction: minor correction

minor typo correction in README.md

bochaco committed:
Merge pull request #360 from safesurfer/minor-correction

style/correction: minor correction

Commit:
feat/webrtc: implement webrtc signalling

Implemented webRTC secure signalling application on SAFE Network.

Commit:
fix/credential: remove credential

Removed credentials

Commit:
Chore/readme: update readme.md

Update readme.md for configuration of STUN and TURN server

Commit:
fix/fetch_publicNames: resolve fetch public name issue

Resolve public name issues

username1001 committed:
Change hosing to hosting (#364)

Commit:
fix/ux: improve UI/UX

Improved UI/UX features of the application.
Removed server details.
Localised texts.

Commit:
fix/disable_logs: disable logs for production

disable logs for production

Commit:
fix/improve_messages: improve label and message texts

Improved label and message texts to make things consistent throughout the application.

Commit:
fix/remove_credentials: remove credentials

Replace place holder instead of credentials.

Commit:
fix/travis: fix travis issue

Fixed travis issue, failing to find test script on root package.json

bochaco committed:
Merge pull request #363 from shankar2105/webrtc

Implement webRTC signalling

safesurfer committed:
updated test suite

safesurfer committed:
upgrade safe-app version to 0.7.0

bochaco commented on email_app/package.json:
```diff
@@ -98,7 +98,7 @@
   },
   "dependencies": {
-    "@maidsafe/safe-node-app": "0.6.0",
+    "@maidsafe/safe-node-app": "0.7.0",
```
I would go for upgrading to very last one which is v0.8.1.

bochaco commented on web_hosting_manager/app/package.json:
```diff
@@ -11,11 +11,11 @@
   "license": "MIT",
   "dependencies": {
-    "@maidsafe/safe-node-app": "0.6.0",
+    "@maidsafe/safe-node-app": "0.7.0",
```
I would go for very latest one which is v0.8.1

hunterlester commented:
Closing, as email app has been archived and web_hosting_manager moved to it's own repository.
